File: src/main/java/com/haozileung/web/servlet/DictionaryServlet.java
/**
 *
 */
package com.haozileung.web.servlet;

import com.google.inject.Inject;
import com.haozileung.infra.pager.PageResult;
import com.haozileung.infra.web.BaseServlet;
import com.haozileung.web.domain.dictionary.Dictionary;
import com.haozileung.web.service.dictionary.IDictionaryService;
import org.apache.commons.beanutils.BeanUtilsBean;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import java.lang.reflect.InvocationTargetException;

/**
 * @author Haozi
 */
@WebServlet(urlPatterns = "/dictionary", loadOnStartup = 1)
public class DictionaryServlet extends BaseServlet {

    private static final Logger logger = LoggerFactory.getLogger(DictionaryServlet.class);
    /**
     *
     */
    private static final long serialVersionUID = 4737169523928049554L;

    @Inject
    private IDictionaryService dictionaryService;

    @Override
    public PageResult<Dictionary> get(HttpServletRequest req, HttpServletResponse resp) {
        Dictionary dictionary = new Dictionary();
        PageResult<Dictionary> result = new PageResult<>();
        try {
            BeanUtilsBean.getInstance().populate(dictionary, req.getParameterMap());
            int pageNo = Integer.parseInt(req.getParameter("pageNo"));
            result = dictionaryService.query(dictionary,pageNo,20);
            return result;
        } catch (IllegalAccessException e) {
            logger.error(e.getMessage(), e);
        } catch (InvocationTargetException e) {
            logger.error(e.getMessage(), e);
        }finally {
            return result;
        }
    }
}

File: src/main/java/com/haozileung/web/controller/DictionaryServlet.kt
/**
 *
 */
package com.haozileung.web.controller;

import com.haozileung.infra.pager.PageResult
import com.haozileung.infra.web.BaseServlet
import com.haozileung.infra.web.Initializer
import com.haozileung.web.domain.dictionary.Dictionary
import com.haozileung.web.service.dictionary.IDictionaryService
import org.apache.commons.beanutils.BeanUtilsBean
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import java.lang.reflect.InvocationTargetException
import javax.servlet.annotation.WebServlet
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

/**
 * @author Haozi
 */
@WebServlet(name = "dictionary", value = "/dictionary",loadOnStartup = 1)
class DictionaryServlet() : BaseServlet() {

    private final val logger:Logger = LoggerFactory.getLogger(DictionaryServlet::class.java);

    val dictionaryService: IDictionaryService = Initializer.instance?.getInstance(IDictionaryService::class.java)!!;

    override fun  get(req:HttpServletRequest, resp:HttpServletResponse) :PageResult<Dictionary>{
        val  dictionary :Dictionary =  Dictionary();
        var result:PageResult<Dictionary>  = PageResult();
        try {
            BeanUtilsBean.getInstance().populate(dictionary, req.parameterMap);
            val offset = Integer.parseInt(req.getParameter("offset"));
            val  limit = Integer.parseInt(req.getParameter("limit"));
            result = dictionaryService.query(dictionary,offset,limit);
        } catch ( e:IllegalAccessException) {
            logger.error(e.message, e);
        } catch ( e:InvocationTargetException) {
            logger.error(e.message, e);
        }finally {
            return result;
        }
    }
}
